CommonTools/RecoAlgos/plugins/JetDeltaRValueMapProducer.cc

```diff
@@ -166,7 +166,7 @@ class JetDeltaRValueMapProducer : public edm::EDProducer {
   const std::vector<std::string>                   valueLabels_;
   const bool                                       lazyParser_;
   bool                                             multiValue_;
-  std::map<std::string, StringObjectFunction<T> >  evaluationMap_;
+  std::map<std::string, std::unique_ptr<const StringObjectFunction<T> > >  evaluationMap_;
```

Comment:
Ah , that mail just sent. OK I have read through your comments. Doesn't
look so bad. Thanks for taking a look!

(Sent from my Nexus 6)
On Aug 8, 2015 6:26 PM, "Lindsey Gray" <lindsey.gray@gmail.com> wrote:

> I thought it was threadsafe *unless* lazy parsing is used.
>
> @VinInn knows more.
>
> (Sent from my Nexus 6)
> On Aug 8, 2015 4:33 PM, "Chris Jones" <notifications@github.com> wrote:
>
>> In CommonTools/RecoAlgos/plugins/JetDeltaRValueMapProducer.cc
>> <https://github.com/cms-sw/cmssw/pull/10620#discussion_r36579187>:
>>
>> > @@ -166,7 +166,7 @@ class JetDeltaRValueMapProducer : public edm::EDProducer {
>> >    const std::vector<std::string>                   valueLabels_;
>> >    const bool                                       lazyParser_;
>> >    bool                                             multiValue_;
>> > -  std::map<std::string, StringObjectFunction<T> >  evaluationMap_;
>> > +  std::map<std::string, std::unique_ptr<const StringObjectFunction<T> > >  evaluationMap_;
>>
>> I do not believe the StringObjectFunction is thread safe. Especially if
>> lazyParser_ is true.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/cms-sw/cmssw/pull/10620/files#r36579187>.
>>
>